static/js/src/public/details/integrations/components/App/App.tsx

```diff
@@ -12,7 +12,13 @@ import { useRecoilState, useRecoilValue } from "recoil";
 import { filterChipsSelector, filterState } from "../../state";
 
 const getIntegrations = async (charm: string): Promise<IInterfaceData[]> => {
-  const resp = await fetch(`/${charm}/integrations.json`);
+  let resp;
+  const selectedChannel = window.location.search.split("channel=")[1];
```

Comment:
It's more text, but it might be more reliable to:
```
const url = new URL(document.location);
const selectedChannel = url.searchParams.get("channel");
```
Generally doing splits etc. on urls will at some point be sad.